README.md

```diff
@@ -37,6 +37,209 @@
+
+		let library = Building(address: “Main Street", height: 20.0, favoriteFood: "Pasta")
+
+		print(description)
+			//Address: Main Street
+			//Height: 20.0
+
+This patterns lets you create all sorts of classes for all sorts of objects. You simply declare the relevant properties and set them in the `init` method. As long as you remember to always implement the code for textDescription, you’ll always be able to get a text version of the data in any of your class instances.
+
+## Customizing textDescription for Each Class with a Protocol
+This is the declaration of `textDescription` that you have to customize each time, but once you do that, you’ll always have it ready when you need to print it out or display it. You just have to remember to declare `textDescription`.
+
+You don’t have to make mental notes or keep a list of reminders for each class you create: you can use a *protocol* to specify that your class must be able to return a `textDescription` property. You still have to write the specific `textDescription` code in each class, but once you’re done that, you can always write
+
+		print (myDog.textDescription)
+		print (myBuilding.textDescription)
+
+and so forth without going back to check on the details of what you’ve written.
+
+Here’s how you specify a protocol that helps you provide this functionality.
+
+There are three steps - declaring the protocol, adopting it in your class declaration, and conforming to it with your class code.
+
+1.	You declare the protocol. In this case, the protocol declares that there will be a textDescription property in your class.
+2. 	You adopt the protocol in your class. 
+3. 	You conform to the protocol. This means that you provide the properties and methods that the protocol requires. In this case, there’s only one property—textDescription
+
+Here’s how you do this.
+
+### 1. Declaring the protocol. 
+The protocol declaration will look like this:
+
+		protocol TextDescription {
+
+    			var textDescription: String { get }
+
+		}
+
+This declares a protocol and gives it a name — `TextClass Description`. The protocol declares that you will implement any and all properties and methods that it declares. In this case, there is only one property: `textDescription`. There can be several, and in some cases, you can make some of them optional but that’s for later on.
+
+> Protocol names are always capitalized just as Class names are. In the code shown here, remember that the protocol named TextDescription (capitalized) contains a variable textDescription that is lower-case. The protocol can contain many variables and even functions (functions in a protocol are for later on). The name of the protocol itself is separate from the names of the variables and functions within it.
+
+### 2.  Adopting the protocol.
+
+In your Class declaration, you *adopt* the protocol. Right now, the class declaration looks like this
+
+		class Person {
+
+Add the protocol so that it reads:
+
+		class Person: TextDescription
+		
```

Comment:
Right at this point, I think we can be slightly more explicit with what's going on here. Something like this:

Now our class `Person` conforms to the protocol `TextDescription`. We are making a promise that we will provide all of the properties and methods declared in the protocol. We know (having created the protocol above) that there is one property we need to implement. In fact, the compiler would yell at us (give us a warning) that we need to implement what it is the protocol wants us to implement if we haven't already done so.